hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -549,6 +551,14 @@ def characters(
       that list will be not be produced. Any overlap between
       ``whitelist_characters`` and ``blacklist_characters`` will raise an
       exception.
+    - ``codec`` can accept `ascii`, `utf-8` or None as an option. While None
+      will not have any extra limitations, `ascii` limit characters to be valid
+      ascii (i.e. codepoints 0 to 127) and `utf-8` will include most characters
+      except the surrogate characters. In another word,
+      `characters(codec="ascii")`` will be equivalent to
+      `characters(min_codepoint=0, max_codepoint=127))` and
+      `characters(codec="utf-8")`` will be equivalent to
+      `characters(blacklist_categories=['Cs']))`.
```

Comment:
I'd omit the implementation details, and try writing it as if codecs _in general_ are supported and link to [these docs](https://docs.python.org/3/library/codecs.html#encodings-and-unicode); then note that currently only ascii and utf-8 are implemented.